### DISCUSSION TITLE
Configurable build server region 🌍

### DISCUSSION THREAD

**Discussion by mountainash** [vercel/vercel#10888] [General]
@vercel-infra is it possible to choose which region the **build process** runs? Given builds can make many API/DB connections to a backend; _build server location_ choice can optimise build times.

Particularly for NextJS sites where a large portion of most sites are statically built in the build process with content pulled from a headless CMS - it makes sense that the build happens near the Backend, [afterwards the output can be CDN'ed around the globe].

There is currently a Vercel project setting to pin a location when **Functions** run, but I can't see anything (Settings, configuration file or Help Docs) about how I can pin **build processes** to a preferred server region.

Can "build processes" be locked to a region?

**Comment by PeterBowater** [comment]
Would like this also, given our data source is immovable in London, and the build happens in Washington, US

**Reply by mountainash** [reply]
Vercel Support told me **all** builds happen in "US based regions" - what happens when those regions are effected by an outage; nobody can release? 🤷 So it would be good for redundancy to also have another DC that can do builds (and even better if that was located in Europe [to begin with]).

**Reply by sarbloc** [reply]
Having issue where an api I use is blocked from US region so my app fails to fetch data on build. This data needs to be fetched on server-side too, so now I'm stuck.

We need build region setting!!

**Reply by mountainash** [reply]
@sukhsinghcodes as a workaround; you can build using another service (outside the US) and then deploy the [built] output to Vercel. Server side (functions) can be set to run in a non-US region. It's not ideal/easiest, but you aren't "stuck".
